bundles/io/org.openhab.io.caldav/src/main/java/org/openhab/io/caldav/internal/job/EventReloaderJob.java

```diff
@@ -403,22 +383,42 @@ public void loadEvents(String filename,
 
     /**
      * Returns a list of categories or an empty list if none found.
+     * 
      * @param vEvent
      * @return
      */
     private List<String> readCategory(VEvent vEvent) {
-        Property propertyCategory = vEvent.getProperty(Property.CATEGORIES);
-        if (propertyCategory != null) {
-            String categories = propertyCategory.getValue();
-            if (categories != null) {
-                String[] categoriesSplit = StringUtils.split(categories, ",");
-                return Arrays.asList(categoriesSplit);
+        PropertyList propertyCategoryList = vEvent.getProperties(Property.CATEGORIES);
+        ArrayList<String> splittedCategoriesToReturn = new ArrayList<String>();
+        if (propertyCategoryList != null) {
+            for (int categorieslinenum = 0; categorieslinenum < propertyCategoryList.size(); categorieslinenum++) {
+                Property propertyCategory = propertyCategoryList.get(categorieslinenum);
+                String categories = propertyCategory.getValue();
+                if (categories != null) {
+                    String[] categoriesSplit = StringUtils.split(categories, ",");
+                    for (String macategorie : categoriesSplit) {
+                        if (!splittedCategoriesToReturn.contains(macategorie)) {
+                            splittedCategoriesToReturn.add(macategorie);
+                        }
+                    }
+                }
             }
-            
         }
-        return new ArrayList<String>();
+        /*
+         * Property propertyCategory = vEvent.getProperty(Property.CATEGORIES);
+         * if (propertyCategory != null) {
+         * String categories = propertyCategory.getValue();
+         * if (categories != null) {
+         * String[] categoriesSplit = StringUtils.split(categories, ",");
+         * return Arrays.asList(categoriesSplit);
+         * }
+         * 
+         * }
+         * return new ArrayList<String>();
+         */
+        return splittedCategoriesToReturn;
```

Comment:
It's better to remove commented out code blocks because it makes the remaining code clearer, and we have the git history to refer to whenever we need to refer to it.  So yes please!